packages/eas-cli/src/build/utils/resourceClass.ts

```diff
@@ -0,0 +1,135 @@
+import { Platform } from '@expo/eas-build-job';
+import { ResourceClass } from '@expo/eas-json';
+import chalk from 'chalk';
+import semver from 'semver';
+
+import { DynamicConfigContextFn } from '../../commandUtils/context/DynamicProjectConfigContextField';
+import { BuildResourceClass } from '../../graphql/generated';
+import Log from '../../log';
+import { ProfileData } from '../../utils/profiles';
+import { getReactNativeVersionAsync } from '../metadata';
+
+const iosResourceClassToBuildResourceClassMapping: Record<ResourceClass, BuildResourceClass> = {
+  [ResourceClass.DEFAULT]: BuildResourceClass.IosDefault,
+  [ResourceClass.LARGE]: BuildResourceClass.IosLarge,
+  [ResourceClass.M1_EXPERIMENTAL]: BuildResourceClass.IosM1Large,
+  [ResourceClass.M1_MEDIUM]: BuildResourceClass.IosM1Medium,
+  [ResourceClass.M1_LARGE]: BuildResourceClass.IosM1Large,
+  [ResourceClass.INTEL_MEDIUM]: BuildResourceClass.IosIntelMedium,
+  [ResourceClass.MEDIUM]: BuildResourceClass.IosMedium,
+};
+
+const androidResourceClassToBuildResourceClassMapping: Record<
+  Exclude<
+    ResourceClass,
+    | ResourceClass.M1_EXPERIMENTAL
+    | ResourceClass.M1_MEDIUM
+    | ResourceClass.M1_LARGE
+    | ResourceClass.INTEL_MEDIUM
+  >,
+  BuildResourceClass
+> = {
+  [ResourceClass.DEFAULT]: BuildResourceClass.AndroidDefault,
+  [ResourceClass.LARGE]: BuildResourceClass.AndroidLarge,
+  [ResourceClass.MEDIUM]: BuildResourceClass.AndroidMedium,
+};
+
+async function resolveIosDefaultRequestedResourceClassAsync(
+  getDynamicProjectConfigAsync: DynamicConfigContextFn,
+  projectDir: string,
+  buildProfile: ProfileData<'build'>
+): Promise<ResourceClass> {
+  const { exp } = await getDynamicProjectConfigAsync({ env: buildProfile.profile.env });
+  const { sdkVersion } = exp;
+  const reactNativeVersion = await getReactNativeVersionAsync(projectDir);
+  if (
+    (sdkVersion && semver.satisfies(sdkVersion, '>=48')) ||
+    (reactNativeVersion && semver.satisfies(reactNativeVersion, '>=0.71.0'))
+  ) {
+    return ResourceClass.M1_MEDIUM;
+  } else {
+    return ResourceClass.DEFAULT;
+  }
+}
+
+function resolveAndroidResourceClass(
+  profileResourceClass?: ResourceClass,
+  resourceClassFlag?: ResourceClass
```

Comment:
Why 2 options? You already handled a case earlier where both values are specified. You should pass here only one option.
Also didn't we decide to deprecate the flag?(I don't remember, just asking)

Reply:
The flag is already deprecated, but we still need to handle it

Reply:
handle yes, but this is not a behavior I would expect from deprecated messages, especially with warnings displayed just before. I'll add a comment there.

Also, the first part of the comment still stands, there is no reason to pass 2 values.